packages/client/src/sync/fetcher/storagefetcher.ts

```diff
@@ -151,26 +152,34 @@ export class StorageFetcher extends Fetcher<JobTask, StorageData[][], StorageDat
    * @returns origin of job is set using either @first property of fetcher or latest hash of partial job
    */
   private getOrigin(job: Job<JobTask, StorageData[][], StorageData[]>): Uint8Array {
-    const { task, partialResult } = job
-    if (task.storageRequests.length > 1 || task.storageRequests[0].first === BIGINT_0) {
-      // peer does not respect origin or limit for multi-account storage fetch
-      return setLengthLeft(bigIntToBytes(BIGINT_0), 32)
-    }
-    const { first } = task.storageRequests[0]!
-    let origin = undefined
-    if (partialResult) {
-      const lastSlotArray = partialResult[partialResult.length - 1]
-      const lastSlot = lastSlotArray[lastSlotArray.length - 1]
-      origin = bigIntToBytes(bytesToBigInt(lastSlot.hash) + BIGINT_1)
-    } else {
-      origin = bigIntToBytes(first + BIGINT_1)
+    let origin: Uint8Array
+
+    try {
```

Comment:
could you add a comment  to code which particular error scenario motivated you to put this in try/catch

Reply:
The parsing of partialResult was resulting in indexing and undefined errors since lastSlot was an array in certain instances instead of a slot object. I will add a comment about it to the code.